[prometheus-adapter] Remove colons from the ClusterRoleBinding resource name

**Describe the bug**

The helm lint (for Helm 3.3) fails for the `prometheus-adapter` Helm chart because of the invalid resource name that is not compliant with [Kubernetes name requirements](prometheus-adapter). The error is the following:

```
templates/custom-metrics-apiserver-auth-delegator-cluster-role-binding.yaml: object name does not conform to Kubernetes naming requirements: "prometheus-adapter:system:auth-delegator"
```

Particularly, it does not like the colons in the resource name.
Helm uses the following regular expression to check the validity of a resource name:

`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`

I believe the name should become: `prometheus-adapter-system-auth-delegator`

**Version of Helm and Kubernetes**:

Helm: **3.3.0**
Kubernetes: **1.18.7**

**Which chart**:

[stable/prometheus-adapter](https://github.com/helm/charts/tree/master/stable/prometheus-adapter)

**What happened**:

`helm lint` command fails

**What you expected to happen**:

`helm lint` command passes

**How to reproduce it** (as minimally and precisely as possible):

Using Helm 3.3 run the following commands:

```bash
helm repo add stable https://kubernetes-charts.storage.googleapis.com
helm pull stable/prometheus-adapter --untar
helm lint stable/prometheus-adapter
```

The output is:

```
==> Linting stable/prometheus-adapter
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/custom-metrics-apiserver-auth-delegator-cluster-role-binding.yaml: object name does not conform to Kubernetes naming requirements: "prometheus-adapter:system:auth-delegator"

Error: 1 chart(s) linted, 1 chart(s) failed
```

--- a/charts/prometheus-adapter/Chart.yaml
+++ b/charts/prometheus-adapter/Chart.yaml
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: prometheus-adapter
-version: 2.5.1
+version: 2.5.2
 appVersion: v0.7.0
 description: A Helm chart for k8s prometheus adapter
 home: https://github.com/DirectXMan12/k8s-prometheus-adapter

--- a/charts/prometheus-adapter/templates/custom-metrics-apiserver-auth-delegator-cluster-role-binding.yaml
+++ b/charts/prometheus-adapter/templates/custom-metrics-apiserver-auth-delegator-cluster-role-binding.yaml
@@ -7,7 +7,7 @@ metadata:
     chart: {{ template "k8s-prometheus-adapter.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
-  name: {{ template "k8s-prometheus-adapter.name" . }}:system:auth-delegator
+  name: {{ template "k8s-prometheus-adapter.name" . }}-system-auth-delegator
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole

